x-pack/plugins/security_solution/public/timelines/store/middlewares/timeline_pinned_event.ts

```diff
@@ -65,17 +64,17 @@ export const addPinnedEventToTimelineMiddleware: (kibana: CoreStart) => Middlewa
           timelineId: timeline.savedObjectId,
         });
 
-        const response: PinnedEventResponse = get('data.persistPinnedEventOnTimeline', result);
-        if (response && response.code === 403) {
+        const response = result.data.persistPinnedEventOnTimeline;
+        if (response && 'code' in response && response.code === 403) {
           store.dispatch(showCallOutUnauthorizedMsg());
         }
 
         refreshTimelines(store.getState());
 
         const currentTimeline = selectTimelineById(store.getState(), action.payload.id);
-        // The response is null in case we unpinned an event.
+        // The response is null or empty in case we unpinned an event.
         // In that case we want to remove the locally pinned event.
-        if (!response) {
+        if (!response || !('code' in response)) {
```

Comment:
Because the response type can be `null` or `{}` (both mean the event was unpinned) we need to check for the existence of `code` to narrow down the type in the `else` case.